### ISSUE TITLE
Option to enable standard multi-select behavior?

### ISSUE THREAD

**Issue by tsujigiri** [JedWatson/react-select#339]
I was wondering if you think it was a good idea to enable standard multi-select behavior in react-select, when the multi prop is set to true. I.e., send the selected values, with the key repeated for each value in the query string: `foo[]=bar&foo[]=baz`. The way this could work is that instead of the hidden text field, a select field is maintained, which mirrors the selected values as selected options. This would have to be hidden using CSS, instead of the `type="hidden"` attribute. It would also mean more overhead in terms of performance. That's why I think not having this as a default but as an option would be better.

**Comment by JedWatson** [comment]
@tsujigiri what benefit would this add? it's good having the hidden field for form submission by default, but I think the more common use-case is having react-select in a React app that interacts with the value via `props` and `state`, rather than a DOM element.

**Comment by bruderstein** [comment]
I agree partially, however I can see the use case if you have a complex form handled from react, and you then want to do a real form-submit post with all the data, it'd be nice if react-select "worked as expected" in that case.

Obviously maintaining the selection in a hidden select is quite a lot of work, for a reasonably unlikely use case, and even if it was default-disabled, maintaining the code could be hairy.

However, I think there's an easier solution :smile: 

If we simply render a hidden input for each selected option, then what gets submitted is identical. We could do this only if  the `name` prop is set, then we're not performing any extra work unless it's necessary.

Code is pretty trivial, I think:
```js
var formValue = null;
if (this.props.name) {
   formValue = this.state.values.map(option => 
        <input type="hidden" name={this.props.value} value={option.value} />)
}
```

Then just use `formValue` instead of the current comma concatenated hidden value.

**Comment by tsujigiri** [comment]
I don't think using an actual select is that much more complicated. Instead of inputs you use options, have them all selected and wrap them in a select:

```javascript
var selectedOptions = this.state.values.map(option => 
  <option style="display: none;" value={option.value} selected="selected" />)
React.createElement('select', { name: this.props.name }, selectedOptions);
```

**Comment by bruderstein** [comment]
That's a very good point. Enabling this by setting the name property seems
like a good way to ensure this isn't done unnecessarily.

I can see both sides of the "do we want to support this" argument though.
